x/stakeibc/keeper/keeper.go

```diff
@@ -260,3 +260,35 @@ func (k Keeper) IsRedemptionRateWithinSafetyBounds(ctx sdk.Context, zone types.H
 	}
 	return true, nil
 }
+
+// Get max number of validators to confirm we won't exceed it when adding a new validator
+// Types of additions:
+//	* change a weight from zero to non-zero
+//  * add a new validator with non-zero weight
+func (k Keeper) ValSetHasSpace(ctx sdk.Context, validators []*types.Validator) (bool, error) {
+
+	// get max val parameter
+	maxNumVals, err := cast.ToIntE(k.GetParam(ctx, types.KeySafetyNumValidators))
+	if err != nil {
+		errMsg := fmt.Sprintf("Error getting safety max num validators | err: %s", err.Error())
+		k.Logger(ctx).Error(errMsg)
+		return false, sdkerrors.Wrap(types.ErrMaxNumValidators, errMsg)
+	}
+
+	// count up the number of validators with non-zero weights
+	numNonzeroWgtValidators := 0
+	for _, validator := range validators {
+		if validator.Weight > 0 {
+			numNonzeroWgtValidators++
+		}
+	}
+
+	// check if the number of validators with non-zero weights is below than the max
+	if numNonzeroWgtValidators >= maxNumVals {
```

Comment:
Since an error is thrown here, there's no need to return a boolean as well. And if you return only an error, then the calling functions only have to check if `err != nil` (i.e., no need to check the `hasSpace` variable as well)

My suggestion would be to remove the boolean and then call this something like `ConfirmValSetHasSpace`

Reply:
Much cleaner approach, nice suggestion! Done.